lib/babe/build.go

```diff
@@ -270,6 +265,18 @@ func (b *BlockBuilder) buildBlockExtrinsics(slot Slot) []*transaction.ValidTrans
 			if _, ok := err.(*DispatchOutcomeError); !ok {
 				continue
 			}
+
+			// don't drop transactions that may be valid in a later block ie.
+			// run out of gas for this block or have a nonce that may be valid in a later block
+			if tvErr, ok := err.(*TransactionValidityError); ok {
```

Comment:
Use `errors.As(err, &e)` https://blog.golang.org/go1.13-errors

Reply:
You can remove indentation
```
	tvErr, ok := err.(*TransactionValidityError)
	if !ok {
		continue
	}
	switch tvErr.msg {
	case exhaustsResourcesMsg, invalidTransactionMsg:
		hash, err := b.transactionState.Push(txn)
		if err != nil {
			logger.Debug("failed to re-add transaction to queue", "tx", hash, "error", err)
		}
	}
```

Reply:
done